spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+describe Api::V1::BooksController do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'when fetching all the books' do
+      let!(:books) { create_list(:book, 3) }
+
+      before do
+        get :index
+      end
+
+      it 'responds with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'when filtering books by genre' do
+      let!(:drama) { create_list(:drama, 3) }
+      let(:books) do
+        arr = create_list(:book, 3)
+        arr.push(*drama)
+      end
```

Comment:
there's an easier way for this. The `create` and `create_list` method support fixed parameters, so you could do
`let(:books) { create_list(:book, 3, genre: 'drama') }` To create 3 different drama books

Reply:
Also, it's better to also create some random books with other random genres to test that the filter actually works. If you only create the drama books the filter could be broken and this test would pass anyway